src/mesh/mesh.hpp

```diff
@@ -147,8 +147,22 @@ class Mesh {
                                        SimTime &tm); // called in main loop
   std::function<void(Mesh *, ParameterInput *, SimTime &)> UserWorkAfterLoop =
       &UserWorkAfterLoopDefault;
-  static void UserWorkInLoopDefault(); // called in main after each cycle
-  std::function<void()> UserWorkInLoop = &UserWorkInLoopDefault;
+  static void UserWorkInLoopDefault(Mesh *, ParameterInput *,
+                                    SimTime const &); // called in main after each cycle
+  std::function<void(Mesh *, ParameterInput *, SimTime const &)> PreStepUserWorkInLoop =
+      &UserWorkInLoopDefault;
+  std::function<void(Mesh *, ParameterInput *, SimTime const &)> PostStepUserWorkInLoop =
+      &UserWorkInLoopDefault;
```

Comment:
I don't think you are missing anything.  I didn't feel the need for separate `Pre`- and `PostStepMeshUserWorkInLoop` because the default is a no-op.  If there is a reason to have different default behaviors, then I would definitely agree that there should be separate defaults.  At a minimum I agree the offending comment should be removed. 

Alternatively, we could initialize to `nullptr` instead of setting he default no-op function. 
 Obviously, this would require some check before calling.  We have this pattern in the `StateDescriptor` but seem to use default no-op functions here.  :shrug:

Reply:
Choose your favorite path. We haven't been consistent.

Reply:
Actually, I think you're right that it is not an issue. It's only `UserWorkInLoopDefault` that is called twice, but the user has no way of overriding that default function. I was concerned that an unsuspecting user would implement that method and then it gets executed twice, but for that to happen, they would have to assign their function to both `PreStepUserWorkInLoop` and `PostStepUserWorkInLoop` and then they would want it to be executed twice per time step. So I think we're all good.